nova/templates/etc/_nova.conf.tpl

```diff
@@ -154,7 +154,13 @@ admin_url = {{.Values.global.keystone_api_endpoint_protocol_admin}}://{{include
 admin_tenant_name={{.Values.global.keystone_service_project}}
 api_endpoint={{.Values.global.ironic_api_endpoint_protocol_internal}}://{{include "ironic_api_endpoint_host_internal" .}}:{{ .Values.global.ironic_api_port_internal }}/v1
 
+{{ if .Values.audit.enabled -}}
 # this is for the cadf audit messaging
 [audit_middleware_notifications]
-driver = messaging
-transport_url = rabbit://{{ .Values.rabbitmq_user | default .Values.global.rabbitmq_default_user }}:{{ .Values.rabbitmq_pass | default .Values.global.rabbitmq_default_pass }}@{{include "rabbitmq_host" .}}:5672/
+# topics = notifications
+driver = messagingv2
+{{- with index .Values "rabbitmq-notifications" }}
+transport_url = rabbit://{{ .users.default.user }}:{{ .users.default.password }}@nova-rabbitmq-notifications:{{ .ports.public }}/
```

Comment:
I get:
```
Error: render error in "staging/charts/openstack/charts/nova/templates/hypervisors/hypervisors_kvm.yaml": template: staging/charts/openstack/charts/nova/templates/hypervisors/hypervisors_kvm.yaml:3:25: executing "staging/charts/openstack/charts/nova/templates/hypervisors/hypervisors_kvm.yaml" at <include "kvm_hypervi...>: error calling include: template: staging/charts/openstack/charts/nova/templates/hypervisors/_kvm-deployment.yaml.tpl:29:31: executing "kvm_hypervisor" at <include (print .Temp...>: error calling include: template: staging/charts/openstack/charts/nova/templates/etc-configmap.yaml:12:3: executing "staging/charts/openstack/charts/nova/templates/etc-configmap.yaml" at <include (print .Temp...>: error calling include: template: staging/charts/openstack/charts/nova/templates/etc/_nova.conf.tpl:163:118: executing "staging/charts/openstack/charts/nova/templates/etc/_nova.conf.tpl" at <.ports.public>: can't evaluate field public in type interface {}```

Is there something, I am missing?

Reply:
I ran into that issue yesterday and never fully understood the root cause

But that should have been fixed with the use of the `conditional` field. I am using Helm 2.7 and I am getting none such errors on staging. Are you using the right branch (hermes_nova) also for cc-regions?

Reply:
I use helm 2.7.2 and not the hermes_nova branch of cc-regions.

But it has to work without using the hermes_nova branch of cc-regions, because otherwise, we cannot deploy with the merged changes in other regions. That is something, I'd like to avoid.

Reply:
no region except staging is changed in branch hermes_nova. So that is not a problem.

Reply:
On the contrary, that is the problem. With this branch, I cannot deploy to staging with master of cc-regions in staging, ergo I cannot deploy in any other region even after merging the changes in cc-regions.

I would rather master to stay deployable.

Reply:
cannot reproduce - reverting to master disables audit successfully also on staging (according to helm diff ...).

Merge?